### PR TITLE
Simplify servlet advice

### PR DESCRIPTION
**Pull request by laurit**

### CODE REVIEW

File: instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java [hunk 1]
```diff
@@ -53,33 +53,22 @@ public static void onEnter(
...
-        scope = updatedContext.makeCurrent();
-      }
+      attachedContext =
+          tracer().updateContext(attachedContext, httpServletRequest, mappingResolver, servlet);
+      scope = attachedContext.makeCurrent();
       return;
     }
 
     Context currentContext = Java8BytecodeBridge.currentContext();
     if (currentContext != null
```
Comment (mateuszrzeszutek) [inline instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java:65]: By the way, `currentContext()` can never return null - you can simplify that bit too 😄

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java (hunk 1 above)
Comment (laurit) [reply instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java]: Thanks, removed it.

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java [hunk 2]
```diff
@@ -53,39 +53,29 @@ public static void onEnter(
...
     }
 
     Context currentContext = Java8BytecodeBridge.currentContext();
-    if (currentContext != null
-        && Java8BytecodeBridge.spanFromContext(currentContext).isRecording()) {
-      // We already have a span but it was not created by servlet instrumentation.
-      // In case it was created by app server integration we need to update it with info from
-      // current request.
+    if (attachedContext != null
+        || Java8BytecodeBridge.spanFromContext(currentContext).isRecording()) {
```
Comment (laurit) [inline instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java:66]: Won't using `isRecording` cause issues when running with sampling?

Comment (trask) [reply instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java:66]: ya, I think `isRecording` should only be used to short-circuit updating a given span's attributes (or name) and probably nothing else, so would support removing this check

**Review by trask — APPROVED** [review]
less code ✅ 
more comments ✅ 
tests pass ✅ 
🚀

File: instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java (hunk 2 above)
Comment (laurit) [reply instrumentation/servlet/servlet-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/servlet/v3_0/Servlet3Advice.java:66]: I'll try to get rid of `isRecording` next week (my initial attempt failed on glassfish and I wasn't able to repro it locally so it requires a bit of debugging). As `isRecording` was already used before this pr I believe we can merge this if there aren't any objections.

**Pull request merged**